### DISCUSSION TITLE
greenlet no longer provides binary wheels for 32-bit win32

### DISCUSSION THREAD

**Discussion by scskrause** [sqlalchemy/sqlalchemy#12022] [Ideas]
The documentation on https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#asyncio-install states the following:

> The asyncio extension requires Python 3 only. It also depends upon the [greenlet](https://pypi.org/project/greenlet/) library. This dependency is installed by default on common machine platforms including:
> 
> `x86_64 aarch64 ppc64le amd64 win32`
> 
> For the above platforms, greenlet is known to supply pre-built wheel files. For other platforms, greenlet does not install by default; the current file listing for greenlet can be seen at [Greenlet - Download Files](https://pypi.org/project/greenlet/#files). Note that there are many architectures omitted, including Apple M1.

However, greenlet hasn't provided any binary wheels for Python > 3.9 on win32 anymore: https://github.com/python-greenlet/greenlet/issues/409#issuecomment-2341259717

So installing SQLAlchemy with 32-bit Python on Windows now requires a C compiler because the greenlet dependency has to be build from source. Shouldn't greenlet be omitted on win32 as well?

**Comment by CaselIT** [comment]
Hi,

Not sure who is still using windows in 32 bits, but feel free to open a PR to remove these

**Reply by scskrause** [reply]
> Not sure how is still using windows in 32 bits

You don't need a 32-bit Windows to run 32-bit Python. You can run 32-bit Python on 64-bit Windows and sometimes you have to, e.g. when you're working with legacy 32-bit ODBC drivers.

**Reply by CaselIT** [reply]
as mentioned we can remove it so that it doesn't install it by default.

**Reply by CaselIT** [reply]
The file to change is this one, in the 2.0 branch https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0/setup.cfg#L42

**Reply by scskrause** [reply]
Unfortunately I can't see a way in PEP 508 to actually detect a 32-bit Python on Windows because even when using a 32-bit interpreter `platform.machine()` still returns `AMD64` when it's run on 64-bit Windows (which I assume is almost always the case).

**Reply by CaselIT** [reply]
not sure if it can be done then.
2.1 will no longer require greenlet by default, but that's not something that can be changed in the 2.0 release.

You should be able to install sqlalchemy without dependencies. Everything (outside asyncio) should work without greenlet
